error_cause_user_initiated_abort.go

```diff
@@ -0,0 +1,32 @@
+package sctp
+
+import (
+	"fmt"
+)
+
```

Comment:
Are you able to add the following text just to be consistent with other error_cause_xxx.go files?
```
/*
   This error cause MAY be included in ABORT chunks that are sent
   because of an upper-layer request.  The upper layer can specify an
   Upper Layer Abort Reason that is transported by SCTP transparently
   and MAY be delivered to the upper-layer protocol at the peer.

        0                   1                   2                   3
        0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
       |         Cause Code=12         |      Cause Length=Variable    |
       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
       /                    Upper Layer Abort Reason                   /
       \                                                               \
       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
*/
```

Reply:
Sure, I'll try to address it over the next couple of days

Reply:
Done!

Reply:
Thanks @jeremija !

Some auto tests are failing with TestAssociation_HandlePacketBeforeInit/Abort.
From what I can see, a.handleAbort() calls a.close() which sets the state to `closed`. Then the test case calls a.Close() which complains "use of closed network connection".

I think calling a.close() on the receipt of Abort chunk. I believe we could skip a.Close() when the test case is `"Abort"` in TestAssociation_HandlePacketBeforeInit.

Otherwise, it all looks good to me!

Reply:
This is something from before but I think this test case name should be changed to something like: "TestAssociation_HandlePacketInCookieWaitState" IMO.

Reply:
Thanks, I didn't realize there was a failing test. It should be fixed now. I also renamed the test case.